File: 0067-add-binary/0067-add-binary.java
class Solution {
    public String addBinary(String a, String b) {
        StringBuilder sb = new StringBuilder();
        int i = a.length() - 1;
        int j = b.length() - 1;
        int carry = 0;
        while (i >= 0 || j >= 0) {
            int sum = carry;
            if (i >= 0) {
                int num1 = a.charAt(i) - '0';
                sum += num1;
            }
            if (j >= 0) {
                int num2 = b.charAt(j) - '0';
                sum += num2;
            }
            carry = sum / 2;
            int print = (sum % 2);
            sb.append((char) (print + '0'));
            i--;
            j--;
        }
        if (carry == 1) {
            sb.append("1");
        }
        sb.reverse();
        return sb.toString();
    }
}

File: 0067-add-binary/0067-add-binary.py
class Solution:
    def addBinary(self, a: str, b: str) -> str:
        i = len(a)-1
        j = len(b)-1
        ans = []
        carry = 0
        while i >= 0 or j >= 0:
            sum = carry
            if i >= 0:
                sum += ord(a[i]) - ord("0")
            if j >= 0:
                sum += ord(b[j]) - ord("0")   
            carry = sum // 2
            ans.append(chr((sum%2)+ord("0")))
            i -= 1
            j -= 1

        if carry == 1:
            ans.append("1")    

        ans.reverse()
        return "".join(ans)

        